### ISSUE TITLE
Assertion failure in redundantSelf rule when using organizeDeclarations

### ISSUE THREAD

**Issue by nicklockwood** [nicklockwood/SwiftFormat#725]
The following test raises an assertion on line 2721 in Rules.swift (`assert(!isTypeRoot)` in the `redundantSelf` rule):

```swift
func testOrganizeDeclarationsAssertionFailureInRedundantSelf() {
    let input = """
    class Foo {
        // MARK: Lifecycle

        public init() {
            self.foo = ["foo"]
        }

        // MARK: Private

        private var foo: [String] = []

        private func bar() {
            // Pulse
            guard let baz = bar else {
                return
            }
        }
    }
    """
    testFormatting(for: input, rule: FormatRules.organizeDeclarations)
}
```

This is due to the `organizeDeclarations` rule mangling the input into:

```swift
class Foo {

    // MARK: Lifecycle

    public init() {
        self.foo = ["foo"]
    }

        // MARK: Internal

        guard let baz = bar else {
            return
        }
    }
    // MARK: Private

    private var foo: [String] = []

    private func bar() {
}
```

Interestingly, changing the `// Pulse` comment to `// Baz` fixes it.

I suspect "Pulse" is being wrongly identified as a misspelled section marker, but since it's not a type-level comment it shouldn't be a section marker candidate anyway, so the root cause is probably something else.

@calda would you mind investigating further?

**Comment by nicklockwood** [comment]
Issue seems to be caused by the organizeDeclarationsRule mangling the code into:

```swift
class Foo {
    // MARK: Lifecycle

    public init() {
        foo = ["foo"]
    }

        // MARK: Internal

        guard let baz = bar else {
            return
        }
    }
    // MARK: Private

    private var foo: [String] = []

    private func bar() {
}
```

**Comment by calda** [comment]
Oops! This one is especially bad, thanks for catching it. Fixed in #726.

It looks like:
 - `organizeDeclarations` was incorrectly trying to validate all comments, not just comments at the top level of a type scope.
 - The edit distance from `// Pulse` to `// Private` was less than 3, so it tried to replace the `// Pulse` comment with a corrected MARK comment.
    - This seems overly permissive, so I updated the edit distance threshold to be proportional to the length of the comment. Now `Pulse` and `Private` are no longer seen as approximately equal.
